ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -393,35 +394,22 @@ private void dumpFunctionMetadata(String dbName, Path dumpRoot) throws SemanticE
     }
   }
 
-  /**
-   *
-   * @param ast
-   * @param dbName
-   * @param tblName
-   * @param dbRoot
-   * @return tbl dumped path
-   * @throws SemanticException
-   */
-  private Path dumpTbl(ASTNode ast, String dbName, String tblName, Path dbRoot) throws SemanticException {
-    Path tableRoot = new Path(dbRoot, tblName);
+  private void dumpTable(ASTNode ast, String dbName, String tblName, Path dbRoot)
+      throws SemanticException {
     try {
-      URI toURI = EximUtil.getValidatedURI(conf, tableRoot.toUri().toString());
       TableSpec ts = new TableSpec(db, conf, dbName + "." + tblName, null);
-
-      ExportSemanticAnalyzer.prepareExport(ast, toURI, ts, getNewReplicationSpec(), db, conf, ctx,
-          rootTasks, inputs, outputs, LOG);
+      TableExport.Paths exportPaths = new TableExport.Paths(ast, dbRoot, tblName, conf);
```

Comment:
Well its generally not a good idea to have everything in a class unless its used by other classes, you can have multiple classes that lead to development of a functionality. code deduplication is one of the design goals, but not the only one, we still want to have objects have behaviors specific to what they are doing. The class is anyways inside the class definition of TableExport itself relating to close proximity of its use.